cms/src/test/java/uk/nhs/digital/apispecs/SwaggerCodeGenOpenApiSpecJsonToHtmlConverterTest.java

```diff
@@ -62,6 +62,25 @@ public void getHtmlForSpec_rendersCompleteOpenApiJsonAsHtml() {
         );
     }
 
+    @Test
+    public void getHtmlForSpec_rendersIncompleteOpenApiJsonAsHtml() {
+
+        // given
+        final String incompleteSpecificationJson = from("oasV3_incomplete_no_components-field.json");
+
+        final String expectedSpecHtml = from("oasV3_complete.html");
+
+        // when
+        final String actualSpecHtml = swaggerCodeGenApiSpecHtmlProvider.htmlFrom(incompleteSpecificationJson);
+
+        // then
+        assertThat(
+            "A complete spec has been rendered as HTML using an incomplete specification customised Swagger CodeGen v3",
```

Comment:
Similar comment to that about the method name: the assertion message could probably be more accurate if it said something like `Specification is rendered with no error when the 'components' field is absent from the source JSON`.